mautic/mautic
# Fix flip array for DC

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Select and boolean fields does not work for email dynamic content properly. This issue bring this commit https://github.com/mautic/mautic/commit/1b48802c2a796fd72f1b5b787cc06c08b9aa6666
There is mismatch with label/values 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create select field with options
- `first` label / `1` value
- `second` label / `2` value

Create boolean field

3. Create email and select code mode
4. On Dynamic Content tab add two condition
- select field with second value (you should see correct labels/values)
- boolean field with no (you should see wrong labels/values - numbers as labels)

5.  Then try save email
6. You should see wrong label/values for select field and wrong label/values for boolean field too
7. After PR should works properly






<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11393"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
